core/components/ui/header/locale-switcher.tsx

```diff
@@ -68,17 +79,13 @@ export const LocaleSwitcher = () => {
   };
 
   const handleOnSubmit = () => {
-    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-    const keys = Object.keys(localeLanguageRegionMap) as LocaleType[];
-
-    const newLocale = keys.find(
-      (key) =>
-        localeLanguageRegionMap[key].language === languageSelected &&
-        localeLanguageRegionMap[key].region === regionSelected,
+    const newLocale = locales.find(
+      (locale) => locale.language === languageSelected && locale.region === regionSelected,
     );
 
     if (newLocale) {
-      router.replace('/', { locale: newLocale });
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+      router.replace('/', { locale: newLocale.id as LocaleType });
```

Comment:
🍹 For another PR, but we should probably include a zod type to ensure `LocaleType`